kafkabp/group_consumer_test.go

```diff
@@ -0,0 +1,169 @@
+package kafkabp
+
+import (
+	"context"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/Shopify/sarama"
+)
+
+func TestGroupConsumer(t *testing.T) {
+	// mock producer side by injecting a mocked consumer group
+	mockProducer := newMockConsumerGroup()
+
+	// test implemenation of a consumer
+	testConsumer := newTestConsumer()
+
+	// inject mocked consumer group
+	gc := &groupConsumer{
+		consumer: mockProducer,
+	}
+
+	// start group consumer in the background and collect all errors
+	errors := make(chan error)
```

Comment:
Since you are sending one message to this channel, I would suggest make it buffered with size 1, so the sender won't get blocked (it would cause the goroutine to hang forever if there's a bug causing there's no reader to consume from this channel, because sender is blocked).